### PR TITLE
Drawer mobile

### LINKED ISSUE #345: improve the swipe able drawer for the mobile
To make the drawer more comfortable for users and more beautiful (layout).

image

A video to explain this feature:
https://dadior.s3.ap-northeast-1.amazonaws.com/Screen%20Recording%202021-12-04%20at%205.17.53%20PM.mp4

---

Some hints:

- Please read our [readme](https://github.com/Greenstand/treetracker-web-map-client#treetracker-web) for more information/guide/tutorial.
- Here is [an engineering book](https://greenstand.gitbook.io/engineering/) in Greenstand.
- To know more about our organization, visit our [website](https://greenstand.org).
- If you want to join the slack community (some resources need the community member's permission), please leave your email address.
- More issues in this project: https://github.com/orgs/Greenstand/projects/26

### PR DESCRIPTION
**Pull request by pierrelstan**
# Description

Fix the swiper drawer for mobile.

Fixes #345

- [x] Bug fix (non-breaking change which fixes an issue)

## Screenshots 
Screen Shot 2022-01-13 at 9 48 50 PM

Screen Shot 2022-01-13 at 9 48 58 PM

### CODE REVIEW

**Comment by dadiorchen** [comment]
@pierrelstan what's the situation of this PR

**Comment by pierrelstan** [comment]
@dadiorchen This one is a little bit challenging. The work is in progress.
For the moment, I didn't figure out a way to implement it yet.

**Comment by pierrelstan** [comment]
@dadiorchen  I'm having issues with my code. I have tried to track the div coordinate of the swipe, but it didn't return the value the way I wanted.
 Do you have some time to rubber duck this?

**Comment by dadiorchen** [comment]
I will try to dig into it when I got time, I was ever reading the code of swiperdrawer by MUI, maybe we need to some hack on it, or copy and build our own one.

**Comment by dadiorchen** [comment]
Previously, I saw google map is doing in the same way which we want pretty well and smoothy

**Comment by dadiorchen** [comment]
Seems an e2e test broken on your PR, can you try to find out what happened? @pierrelstan and if it's hard, please be sure to speak here, we will help you out.

**Comment by pierrelstan** [comment]
The next issue that I found is: The children elements within SwipeableDrawer can't scroll because of on-touch move and Mouseover events.

**Comment by dadiorchen** [comment]
@pierrelstan so we can not scroll down the content in the drawer? if it's true, then we need fix it, right?

**Comment by pierrelstan** [comment]
@dadiorchen  Yes.

I'm working on it to fix it.

File: src/components/Drawer.js [hunk 1]
```diff
@@ -1,142 +1,178 @@
...
+          zIndex: 900,
+        }}
+        ModalProps={{
+          keepMounted: true,
+        }}
       >
-        <Box
+        <StyledBox
           sx={{
-            /*height: (t) => `calc(100vh - ${t.spacing(18)})`,*/
```
Comment (bikkimahato) [inline src/components/Drawer.js:103]: Commented code should not be pushed.

File: src/components/Drawer.js [hunk 2]
```diff
@@ -1,142 +1,178 @@
...
+      setHasHeight(hasHeight);
+    }
+  }, [hasHeight]);
+
+  const handleTouch = (event) => {
+    event.stopPropagation();
+
+    const touches = event.targetTouches && event.targetTouches[0];
 
-  // console.log("Tokens List", tokens)
```
Comment (bikkimahato) [inline src/components/Drawer.js:83]: Commented code should not be pushed.

Comment (dadiorchen) [reply src/components/Drawer.js:82]: Thanks for the reivew @bikkimahato

**Comment by dadiorchen** [comment]
@pierrelstan take a look at this: https://github.com/Greenstand/treetracker-web-map-client/issues/406
it might be helpful to solve your problem. we just need to fix the head and scroll the area under it.

**Comment by pierrelstan** [comment]
> @pierrelstan take a look at this: #406 it might be helpful to solve your problem. we just need to fix the head and scroll the area under it.

Ok great , Thank you.

**Comment by dadiorchen** [comment]
@pierrelstan It seems good, let me merge your code first and I will check it later in my branch, if there are problem, let's raise new issue on it!

**Comment by dadiorchen** [comment]
@pierrelstan thanks for you great work!

**Comment by pierrelstan** [comment]
> @pierrelstan It seems good, let me merge your code first and I will check it later in my branch, if there are problem, let's raise new issue on it!

Thank y

**Comment by pierrelstan** [comment]
> @pierrelstan thanks for you great work!

Thank you.

**Pull request merged**